src/util/light_map.h

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************\
+
+Module: Memory-efficient unordered_map
+
+Author: Diffblue ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_LIGHT_MAP_H
+#define CPROVER_UTIL_LIGHT_MAP_H
+
+#include <memory>
+#include <vector>
+#include <unordered_map>
+#include "any_ref.h"
+
+/// Copied data is shared with previous instance to avoid copying of elements
+/// For example having a map with 1000 elements, passing it into the
+/// function and modifying one element
+template<typename key_t, typename value_t>
```

Comment:
`keyT`, `valueT`

Reply:
But we're not using camel case anywhere else in the code.

Reply:
An argument in favor of the camelCase version is that the `*_t` naming convention is reserved for POSIX types